tests/integration-tests/tests/common/schedulers_common.py

```diff
@@ -195,14 +201,18 @@ def assert_job_submitted(self, sbatch_output):  # noqa: D102
         assert_that(match).is_not_none()
         return match.group(1)
 
-    def submit_command(self, command, nodes=1):  # noqa: D102
-        return self._remote_command_executor.run_remote_command("sbatch -N {0} --wrap='{1}'".format(nodes, command))
+    def submit_command(self, command, nodes=1, host=None):  # noqa: D102
+        submission_command = "sbatch -N {0} --wrap='{1}'".format(nodes, command)
+        if host:
+            submission_command += " --nodelist={0}".format(host)
+        return self._remote_command_executor.run_remote_command(submission_command)
 
-    def submit_script(self, script, nodes=1):  # noqa: D102
+    def submit_script(self, script, nodes=1, host=None):  # noqa: D102
         script_name = os.path.basename(script)
-        return self._remote_command_executor.run_remote_command(
-            "sbatch -N {0} {1}".format(nodes, script_name), additional_files=[script]
-        )
+        submission_command = "sbatch -N {0} {1}".format(nodes, script_name)
+        if host:
+            submission_command += " --nodelist={0}".format(host)
```

Comment:
does this work if we append the --nodelist after the scriptname?

Reply:
it works because the integration test works, but I changed it to be aligned with the documentation: https://slurm.schedmd.com/sbatch.html